go/pkg/client/cas.go

```diff
@@ -173,24 +214,16 @@ func getUnifiedMetadata(metas []*requestMetadata) *requestMetadata {
 		actionIDs[m.actionID] = true
 		invocationIDs[m.invocationID] = true
 	}
-	var tnKeys, aiKeys, iiKeys []string
-	for toolName := range toolNames {
-		tnKeys = append(tnKeys, toolName)
-	}
-	for actionID := range actionIDs {
-		aiKeys = append(aiKeys, actionID)
-	}
-	for invocationID := range invocationIDs {
-		iiKeys = append(iiKeys, invocationID)
-	}
-	sort.Strings(tnKeys)
-	sort.Strings(aiKeys)
-	sort.Strings(iiKeys)
-	return &requestMetadata{
-		toolName:     strings.Join(tnKeys, ","),
-		actionID:     strings.Join(aiKeys, ","),
-		invocationID: strings.Join(iiKeys, ","),
+	m := &requestMetadata{
+		toolName:     getUnifiedLabel(toolNames),
+		actionID:     getUnifiedLabel(actionIDs),
+		invocationID: getUnifiedLabel(invocationIDs),
 	}
+	// We cap to a bit less than the maximum header size in order to allow
+	// for some proto fields serialization overhead.
+	const kDefaultMaxHeaderSize = 8 * 1024
```

Comment:
I'd make it a global package variable rather than a function one - the compiler is probably smart enough to inline it, but still :)

Reply:
Done.